src/gpu/generic/sycl/rnn/ref_rnn.hpp

```diff
@@ -0,0 +1,168 @@
+/*******************************************************************************
+* Copyright 2019-2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_GENERIC_SYCL_RNN_REF_RNN_HPP
+#define GPU_GENERIC_SYCL_RNN_REF_RNN_HPP
+
+#include <stdio.h>
+
+#include "common/c_types_map.hpp"
+#include "common/primitive.hpp"
+#include "common/primitive_desc_iterator.hpp"
+#include "common/utils.hpp"
+#include "gpu/generic/sycl/rnn/rnn_utils.hpp"
+#include "gpu/generic/sycl/sycl_gpu_primitive.hpp"
+#include "gpu/gpu_rnn_pd.hpp"
+
+#include "gpu/generic/sycl/sycl_gpu_primitive.hpp"
+#include "gpu/generic/sycl/sycl_primitive_conf.hpp"
+
+#include "gpu/generic/sycl/sycl_gpu_kernel.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace generic {
+namespace sycl {
+
+enum gemm_kind_t { gemm_iter_fwd, gemm_layer_fwd };
+
+template <prop_kind_t aprop>
+struct _ref_rnn_common_t : public primitive_t {
```

Comment:
Templating on `prop_kind_t` doesn't seem like it accomplishes much. Would it be reasonable to remove it?

Reply:
Agreed, it can be introduced when/if the RNN functionality requires it in another PR. Removed from here.